autolab/Autolab
# Category Allowlist for Metrics

## Description
1. Added the endpoints to retrieve category allowlist and to update watchlist configurations.
2. Take category allowlist into consideration when calculating watchlist instances for a course.

## Motivation and Context
To improve the metrics feature.

## How Has This Been Tested?
`rails s`
Testing the endpoints:
1. Test `get_category_allowlist_for_course` by directly visiting "<LOCALHOST>/courses/<COURSENAME>/metrics/get_category_allowlist_for_course".
2. Test `update_watchlist_configuration_for_course` by using Postman or `fetch`. See documentation in file for appropriate parameters. After updating the configuration, use the get API to verify.

Testing watchlist instances:
1. Reset category allowlist to empty.
2. Test the four conditions individually. For each, do the following:
- Set the condition. Refresh the watchlist and verify that the instances contain violation information pertaining to a particular assessment category.
- Use the update endpoint to set category allowlist to contain a relevant category.
- Refresh the watchlist page and verify that watchlist instances no longer contain violation information related to that category.
- Reset the condition to none and refresh the watchlist to empty.
- Set the condition similar to earlier. Verify that the watchlist instances still contain no violation information of the allowlisted category.
3. Try some combination of the four conditions and different allowlist and verify that things look right.

## Types of changes


## Checklist:


## Other issues / help required




If unsure, feel free to submit first and we'll help you along.
